src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/database/jpa/entity/ActionPlanEntity.kt

```diff
@@ -0,0 +1,47 @@
+package uk.gov.justice.digital.hmpps.educationandworkplanapi.app.database.jpa.entity
+
+import jakarta.persistence.Column
+import jakarta.persistence.Entity
+import jakarta.persistence.GeneratedValue
+import jakarta.persistence.Id
+import jakarta.persistence.JoinColumn
+import jakarta.persistence.OneToMany
+import jakarta.persistence.Table
+import jakarta.validation.constraints.NotNull
+import org.hibernate.Hibernate
+import org.hibernate.annotations.UuidGenerator
+import java.util.UUID
+
+@Table(name = "action_plan")
+@Entity
+class ActionPlanEntity(
+  @Id
+  @GeneratedValue
+  @UuidGenerator
+  @NotNull
+  var id: UUID? = null,
+
+  @Column(updatable = false)
+  @NotNull
+  var prisonNumber: String? = null,
+
+  @OneToMany
+  @JoinColumn(name = "action_plan_id")
+  @NotNull
+  var goals: MutableList<GoalEntity>? = null,
```

Comment:
I know we want to get this merged, so we can address this afterwards, but I have a feeling we found a way to avoid having to set the entity fields to `null`.

Reply:
Oh, that would be good